### DISCUSSION TITLE
Migrate files to new folder?

### DISCUSSION THREAD

**Discussion by BlankSlate56** [jonaswinkler/paperless-ng#1611] [Support]
When I first started paperless, I had a different folder structure than I want to have now. I have 2000 files currently loaded into paperless. How can I move those files to a new location without breaking their database link and keep the paperless tags, names, etc?  Note that I know how to change the target folder, I just don't know how to migrate my legacy files.

**Comment by mpfl** [comment]
Caveat: I use Docker.

I just successfully moved ~1000 files to a new NAS using rsync and updated the mount. Paperless didn't bat an eyelid.

The media directory is just a bunch of files. As long as you preserve the structure and file naming conventions, the links to the metadata in the database should be maintained.

**Reply by BlankSlate56** [reply]
I'm using docker on a synology NAS.

"As long as you preserve the structure and file naming conventions,"

The file structure will be different.  Instead of /NAS/docker/paperless/media/originals, I just want /NAS/Records.

Will the rsync method you mention work for this?  Getting Paperless up and running was already stretching my skills.  Do you happen to have any documentation for the rsync method you mention?

**Reply by mpfl** [reply]
I'm not sure that moving originals without moving the other parts of media is a good idea. Moving all of /NAS/docker/paperless/media to /NAS/Records should be possible (unless Synology imposes restrictions for docker bind mounts). My NAS is an Ubuntu Linux server.

rsync is a command line tool that is useful for copying files in a rigorous fashion. I especially like using it when copying/moving files from server to server (which is what I did in this case).

My process would be:

1. Stop the paperless container
2. _Copy_ the contents of media to the new location by whatever means Synology lets you
3. Update the container definition so that the media mount points to /NAS/Records rather than /NAS/docker/paperless/media. Keep a copy of the old container definition.
4. Start the container to see if all your files are still there. If they are, you can consider deleting your files from the original location.

**Reply by BlankSlate56** [reply]
So you're saying that all I really have to do is copy the files over, point paperless to the new location, and restart paperless?  I was hoping that's all I'd need to do but wanted to make sure.

I honestly don't really understand the point of the archive folder.  To me its just creating duplicate copies that I don't need.  As for the thumbnails, as long as paperless knows where those are stored, they shouldn't have to be folder-adjacent to the original folder either?

**Reply by mpfl** [reply]
That's all I just did, albeit moving the files between servers.

When paperless imports a document, it processes the PDF (performs OCR, for example). Since it is modifying the file, it keeps the original file as well (counterintuitively in the `archive` directory). Paperless has had issues with at least one PDF I've imported, so I'm glad it keeps the originals.

Having a look at the same document in both archive and originals, it looks like the file in `originals` is around 15% larger than its counterpart in `archive`.

```
mpfl@mace:/redundant/records/documents$ ls originals/0000001.pdf archive/0000001.pdf -al
-rw-rw---- 1 records records 110693 Feb 10 05:19 archive/0000001.pdf
-rw-rw---- 1 records records 126812 Feb 10 05:19 originals/0000001.pdf
```

I guess you could keep the `originals` directory elsewhere if you bind mount `media` and `media/originals` separately. It's not something I've considered doing, so I haven't done it to confirm that it will work.

**Reply by BlankSlate56** [reply]
ahhh...  So Originals is the OCR'd document while Archive is the actual original non-OCR'd file?  I never could keep that distinction clear despite reading the docs.  I'm just trying to keep my folder hierarchy to a minimum.  I frequently link documents into my bookkeeping software that uses windows explorer.  Having to navigate through multiple levels really gets old.

**Reply by mpfl** [reply]
Actually, I'm wrong. I just checked an invoice I was emailed. `originals` is the file before paperless-ng processes it, and `archive` is post-processing.

I was mistaken because I assumed the larger file was the processed file.
